tracing-subscriber/src/fmt/mod.rs

```diff
@@ -444,23 +441,26 @@ where
     /// Install this Subscriber as the global default if one is
     /// not already set.
     ///
-    /// If the `tracing_log` feature is enabled, this will also install
+    /// If the `tracing-log` feature is enabled, this will also install
     /// the LogTracer to convert `Log` records into `tracing` `Event`s.
     ///
     /// # Errors
     /// Returns an Error if the initialization was unsuccessful, likely
     /// because a global subscriber was already installed by another
     /// call to `try_init`.
     pub fn try_init(self) -> Result<(), impl Error + Send + Sync + 'static> {
-        #[cfg(feature = "tracing_log")]
+        #[cfg(feature = "tracing-log/std")]
```

Comment:
Note: I'm not sure why the `/std` is needed. Without it, it runs into the following error. However, I thought the whole crate assumes `std`, so I'm a bit confused by that.

```
error[E0277]: `?` couldn't convert the error to `std::boxed::Box<tracing_core::dispatcher::SetGlobalDefaultError>`
   --> tracing-subscriber/src/fmt/mod.rs:453:57
    |
453 |         tracing_log::LogTracer::init().map_err(Box::new)?;
    |                                                         ^ the trait `std::convert::From<std::boxed::Box<log::SetLoggerError>>` is not implemented for `std::boxed::Box<tracing_core::dispatcher::SetGlobalDefaultError>`
    |
    = note: the question mark operation (`?`) implicitly performs a conversion on the error value using the `From` trait
    = help: the following implementations were found:
              <std::boxed::Box<(dyn std::error::Error + 'a)> as std::convert::From<E>>
              <std::boxed::Box<(dyn std::error::Error + 'static)> as std::convert::From<&str>>
              <std::boxed::Box<(dyn std::error::Error + 'static)> as std::convert::From<std::borrow::Cow<'a, str>>>
              <std::boxed::Box<(dyn std::error::Error + 'static)> as std::convert::From<std::string::String>>
            and 16 others
    = note: required by `std::convert::From::from`

error: aborting due to previous error

For more information about this error, try `rustc --explain E0277`.
```

Reply:
I would probably just add the `std` feature in the `tracing-log` dependency in this crate's `Cargo.toml.`

Reply:
The standard library is required for the `std::error::Error` impls, as that trait isn't defined in `core`.

Reply:
If I get rid of the 2nd commit in this PR and put this in the `Cargo.toml`:
```toml
tracing-log = { path = "../tracing-log", version = "0.1", optional = true, default-features = false, features = ["std", "log-tracer"] }
```

I get the error:
```
tracing git:es-fix-init ❯ cargo check                                                                                                  ═
error: failed to select a version for `tracing-log`.
    ... required by package `tracing-subscriber v0.1.5 (/home/emschwartz/code/tracing/tracing-subscriber)`
versions that meet the requirements `= 0.1.1` are: 0.1.1

the package `tracing-subscriber` depends on `tracing-log`, with features: `std` but `tracing-log` does not have these features.


failed to select a version for `tracing-log` which could resolve this conflict
```

Am I missing something obvious?

(In case this doesn't work for some reason, the 2nd commit in this PR switches it to use an Enum instead of a Box)

Reply:
Hmm, it's possible `tracing-log` doesn't expose that feature, I don't remember off the top of my head. A `std` feature on `tracing-log` that enables `log/std` should be added if it doesn't already exist.